plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -1978,11 +1978,16 @@ So if getMinSpeed() returns null we fall back to getSpeed().
         final SerialDef serial = new SerialDef("pty", null, (short)0);
         devices.addDevice(serial);
 
+        /* Add a VirtIO channel for SystemVMs for communication and provisioning */
         if (vmTO.getType() != VirtualMachine.Type.User) {
-            final VirtioSerialDef vserial = new VirtioSerialDef(vmTO.getName(), null);
-            devices.addDevice(vserial);
+            devices.addDevice(new ChannelDef(vmTO.getName() + ".vport", ChannelDef.ChannelType.UNIX,
+                                             "/var/lib/libvirt/qemu/" + vmTO.getName() + ".agent"));
         }
 
+        /* Add a VirtIO channel for the Qemu Guest Agent tools */
+        devices.addDevice(new ChannelDef("org.qemu.guest_agent.0", ChannelDef.ChannelType.UNIX,
+                                         "/var/lib/libvirt/qemu/" + vmTO.getName() + ".org.qemu.guest_agent.0"));
```

Comment:
We're hardcoding channel type `unix` here, is that the default or should we allow users to choose between unix, serial and others? Also, about the guest agent name, while `/var/lib/libvirt/qemu` is the default path on most distros we should make it configurable or make it detect the path based on the env.